utils/s3/client.hh

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include <seastar/core/sstring.hh>
+#include <seastar/core/shared_ptr.hh>
+#include <seastar/http/client.hh>
+
+using namespace seastar;
+
+namespace s3 {
+
+struct range {
+    uint64_t off;
+    size_t len;
+};
+
+class client : public enable_shared_from_this<client> {
+    sstring _host;
+    http::experimental::client _http;
+
+    struct private_tag {};
+    future<std::vector<temporary_buffer<char>>> get_object(sstring object_name, std::optional<std::pair<uint64_t, size_t>> range);
+
+public:
+    explicit client(ipv4_addr addr, private_tag);
+    static shared_ptr<client> make(ipv4_addr addr);
+
+    future<uint64_t> get_object_size(sstring object_name);
```

Comment:
can be made const.

Reply:
It looks like it cannot :( It calls `_http.make_request` inside which is not const and compiler complains. The latter, in turn, may start creating new connection to the server thus modifying the "connections pool" and cannot be marked `const`

Reply:
We can mark _http mutable, if we think it's important. _http is like a pointer to the remote server.